pandas/core/case_when.py

```diff
@@ -0,0 +1,207 @@
+                    )
+        if cond_indices:
+            for left, right in zip(cond_indices, cond_indices[1:]):
+                if not left.index.equals(right.index):
+                    raise AssertionError(
+                        "All condition objects must have the same index."
+                    )
+            if replacement_indices:
+                if not replacement_indices[0].index.equals(cond_indices[0].index):
+                    raise AssertionError(
+                        "All replacement objects and condition objects "
+                        "should have the same index."
+                    )
+        else:
+            conditions = [
+                np.asanyarray(cond) if not hasattr(cond, "shape") else cond
+                for cond in conditions
+            ]
+            cond_length = {len(cond) for cond in conditions}
+            if len(cond_length) > 1:
+                raise ValueError("The boolean conditions should have the same length.")
+            cond_length = len(conditions[0])
+            if not is_scalar(default):
+                if len(default) != cond_length:
+                    raise ValueError(
+                        "length of `default` does not match the length "
+                        "of any of the conditions."
+                    )
+            if not replacement_indices:
+                for num, replacement in enumerate(replacements):
+                    if is_scalar(replacement):
+                        continue
+                    if not hasattr(replacement, "shape"):
+                        replacement = np.asanyarray(replacement)
+                    if len(replacement) != cond_length:
+                        raise ValueError(
+                            f"Length of condition{num} does not match "
+                            f"the length of replacement{num}; "
+                            f"{cond_length} != {len(replacement)}"
+                        )
+        if cond_indices:
+            default_index = cond_indices[0].index
+        elif replacement_indices:
+            default_index = replacement_indices[0].index
+        else:
+            default_index = range(cond_length)
+        default = Series(default, index=default_index, dtype=common_dtypes)
+    counter = reversed(range(len(conditions)))
+    for position, condition, replacement in zip(
+        counter, conditions[::-1], replacements[::-1]
+    ):
+        try:
+            default = default.mask(
+                condition, other=replacement, axis=0, inplace=False, level=None
+            )
+        except Exception as error:
+            raise ValueError(
+                f"Failed to apply condition{position} and replacement{position}."
+            ) from error
+    return default
+
+
+def validate_case_when(args: tuple) -> tuple:
+    """
+    Validates the variable arguments for the case_when function.
+    """
+    if not len(args):
```

Comment:
Shouldn't you check at runtime if `args` is a `tuple` ?

Reply:
it is a variable argument, do we still need to check the type, since it should be a tuple?